--- object_detection_demo.py	2021-01-12 10:48:01.926644033 +0800
+++ object_detection_demo_changed.py	2021-01-12 10:50:08.666649762 +0800
@@ -37,6 +37,8 @@
 logging.basicConfig(format='[ %(levelname)s ] %(message)s', level=logging.INFO, stream=sys.stdout)
 log = logging.getLogger()
 
+import os
+os.environ["OPENCV_FFMPEG_CAPTURE_OPTIONS"] = "protocol_whitelist;file,rtp,udp,rtmp,tcp"
 
 def build_argparser():
     parser = ArgumentParser(add_help=False)
@@ -230,6 +232,28 @@
         log.error('OpenCV: Failed to open capture: ' + str(input_stream))
         sys.exit(1)
 
+    ret, frame = cap.read()
+    height, width, ch = frame.shape
+    dimension = '{}x{}'.format(width, height)
+    fps = str(cap.get(cv2.CAP_PROP_FPS))
+    command = ['ffmpeg',
+    '-y',
+    '-f', 'rawvideo',
+    '-use_wallclock_as_timestamps', '1',
+    '-vcodec','rawvideo',
+    '-s', dimension,
+    '-pix_fmt', 'bgr24',
+    '-r', fps,
+    '-i', 'pipe:0',
+    '-c:v', 'libx264',
+    '-pix_fmt', 'yuv420p',
+    '-preset', 'ultrafast',
+    '-f', 'flv',
+    'rtmp://127.0.0.1/live/out.flv']
+
+    import subprocess as sp
+    proc = sp.Popen(command, stdin=sp.PIPE, shell=False)
+
     next_frame_id = 0
     next_frame_id_to_show = 0
 
@@ -260,7 +284,8 @@
             frame = draw_detections(frame, objects, palette, model.labels, args.prob_threshold, has_landmarks)
             metrics.update(start_time, frame)
             if not args.no_show:
-                cv2.imshow('Detection Results', frame)
+                #cv2.imshow('Detection Results', frame)
+                proc.stdin.write(frame.tobytes())
                 key = cv2.waitKey(1)
 
                 ESC_KEY = 27
@@ -306,7 +331,8 @@
             frame = draw_detections(frame, objects, palette, model.labels, args.prob_threshold, has_landmarks)
             metrics.update(start_time, frame)
             if not args.no_show:
-                cv2.imshow('Detection Results', frame)
+                #cv2.imshow('Detection Results', frame)
+                proc.stdin.write(frame.tobytes())
                 key = cv2.waitKey(1)
 
                 ESC_KEY = 27
